contrib/kotlin-validator.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+ # Kiwix Android
+ # Copyright (c) 2022 Kiwix <android.kiwix.org>
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program. If not, see <http://www.gnu.org/licenses/>.
+ #
+ #
+
+ if ./gradlew buildKotlinToolingMetadata; then
+   version=$(< app/build/kotlinToolingMetadata/kotlin-tooling-metadata.json  tr { '\n' | tr , '\n' | tr } '\n' | grep "buildPluginVersion" | awk  -F'"' '{print $4}')
+   if [ $version == "1.7.0" ];
+   then
+    echo "Build System is using kotlin Version 1.7.0"
+   else
+     echo "Build Failure : Due to kotlin Version 1.7.0 is missing"
```

Comment:
@kelson42 I think we only checking the version, not the features.

Reply:
I see indeed no value in this.

Reply:
hi @kelson42 I misunderstood the requirement here as you quoted "I see nothing in the CI checking that we really use Kotlin 1.7.0 and complain otherwise. Why?" on  https://github.com/kiwix/kiwix-android/pull/2884  , i thought we have to make sure we are using kotlin verison 1.7.0 so I was running ```buildKotlinToolingMetadata``` which actually build a kotlin tooling report and saved information on json file and from shell script i am fetching the version and placed a check.

Reply:
hi @kelson42 ,

I already do refactoring of code to support latest 1.7.0 on this https://github.com/kiwix/kiwix-android/pull/2884, please let me know what actually CI really needs to report here

Reply:
Inwant a red CI if someone introduces non kotlin 1.7.p code via a PR.

Reply:
>I want a red CI if someone introduces non kotlin 1.7.0 code via a PR. The CI should clearly state that is is a Kotlin style problem and allow to easily point to the line where the problem is...

@kelson42 I think it's unnecessary. Languages evolve all the time especially kotlin. We'll pretty soon update to kotlin 1.8. Is it necessary to give attention to the style of the Language or am I missing something? @MohitMaliFtechiz any thought?

Reply:
Yes it is. It is an easy solution to improve overall code quality and coherency... without much effort.

If Kotlin continues to improve over time, then we will dedide to follow the new standart... or not.

Reply:
Yes it is. It is an easy solution to improve overall code quality and coherency... without much effort.

If Kotlin continues to improve over time, then we will dedide to follow the new standart... or not.

Reply:
hi @gouri-panda , We can update Klint , don't you think that also doing the same job.

Reply:
Yes! There are a lot of tools available for do this kind of job.

Reply:
If we want a specific code style in our codebase, we can configure it in our project to do this. `Deket` tool is convenient here to do this kind of stuff.